debian/freemind.1
.\"                                      Hey, EMACS: -*- nroff -*-
.\" First parameter, NAME, should be all caps
.\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
.\" other parameters are allowed: see man(7), man(1)
.TH FREEMIND 1 "February  9, 2004"
.\" Please adjust this date whenever revising the manpage.
.\"
.\" Some roff macros, for reference:
.\" .nh        disable hyphenation
.\" .hy        enable hyphenation
.\" .ad l      left justify
.\" .ad b      justify to both left and right margins
.\" .nf        disable filling
.\" .fi        enable filling
.\" .br        insert line break
.\" .sp <n>    insert n+1 empty lines
.\" for manpage-specific macros, see man(7)
.SH NAME
freemind \- A Java Program for creating and viewing Mindmaps

.SH SYNOPSIS
.B freemind
.I <no_options>

.SH DESCRIPTION
This manual page documents briefly the
.B freemind
command.
This manual page was written for the Debian distribution
because the original program does not have a manual page.
Instead, it has documentation as a mindmap (see help menu).
.PP
.\" TeX users may be more comfortable with the \fB<whatever>\fP and
.\" \fI<whatever>\fP escape sequences to invode bold face and italics, 
.\" respectively.
\fBfreemind\fP is a program that allows to create and view so-called
Mind Maps, as well as export them to HTML.

.SH OPTIONS
This program does not have any option. Just start it!

.SH ENVIRONMENT
.IP \fBDEBUG\fP
If this environment variable is set, freemind gives more information on how it
finds its different elements.
.IP \fBJAVACMD\fP
Default value for the java virtual machine (the \fBjava\fP command).
.IP \fBJAVA_BINDIR\fP
If $\fBJAVACMD\fP is not set, the java virtual machine is searched under
$\fBJAVA_BINDIR\fP\fI/java\fP.
.IP \fBJAVA_HOME\fP
If $\fBJAVACMD\fP and $\fBJAVA_BINDIR\fP are not set, the java virtual machine
is searched under $\fBJAVA_HOME\fP\fI/bin/java\fP.
.IP \fBPATH\fP
If all $\fBJAVA\fP... environment variables fail, the java virtual machine
command (\fBjava\fP) is searched in the $\fBPATH\fP (with \fBwhich\fP).
.IP \fBCLASSPATH\fP
.IP \fBADD_JARS\fP
The $\fBCLASSPATH\fP variable also lets one specify additional jars,
which is good, if you want to add a new Look&Feel jar (the motif one is so
ugly...). Alternatively, the content of $\fBADD_JARS\fP will be prepended to
$\fBCLASSPATH\fP.

.SH FILES
.IP \fI/etc/freemind/freemindrc\fP
.IP \fI~/.freemind/freemindrc\fP
.B freemind
tries to read its environment variables from \fI/etc/freemind/freemindrc\fP
and \fI~/.freemind/freemindrc\fP, so that specific settings are possible at
machine and user level.
.IP \fI/usr/bin/java\fP
If all searches through environment variables fail, the java virtual machine
is expected to be \fI/usr/bin/java\fP;
if not, \fBfreemind\fP exits with an error.

.SH SEE ALSO
The online documentation under
.UR http://freemind.sf.net/
.I http://freemind.sf.net/
.UE

.SH AUTHOR
This manual page was written by Eric Lavarde <deb@zorglub.s.bawue.de>,
for the Debian project (but may be used by others).


debian/changelog
ossxp-freemind (0.9.0~rc6-0.17.git~mmx) unstable; urgency=low

  * Upstream upgrade to 0.9.0_RC_6.

 -- Jiang Xin <worldhello.net@gmail.com>  Sat, 16 Jan 2010 23:43:38 +0800

ossxp-freemind (0.9.0~rc1-0.17.git~mmx) unstable; urgency=low

  * Maintain patches using git + topgit.

 -- Jiang Xin <worldhello.net@gmail.com>  Fri, 15 Jan 2010 11:10:48 +0800

ossxp-freemind (0.9.0~rc1-0.17~mmx) unstable; urgency=low

  * Fixed #18: Mindmap saved in utf-8 when export as flash.

 -- Jiang Xin <worldhello.net@gmail.com>  Wed, 25 Feb 2009 18:29:03 +0800

ossxp-freemind (0.9.0~rc1-0.16~mmx) unstable; urgency=low

  * Create a project on sf.net: freemind-mmx;
  * Upstream upgrade to 0.9.0_RC_1.
  * New version format compatible with apt;
  * Bugfix: Subversion switch to Hg not completely, cause losing 
    wh_nonascii_in_utf8 option in freemind.properties.
  * Translate zh_CN resources for 0.9.0rc1.

 -- Jiang Xin <worldhello.net@gmail.com>  Mon, 16 Feb 2009 01:35:05 +0800

ossxp-freemind (0.9.0~b20-15~wh) unstable; urgency=low

  * Convert from Subversion to Mercurial;
  * Maintain patches using MQ;

 -- Jiang Xin <worldhello.net@gmail.com>  Sat, 27 Dec 2008 16:29:24 +0800

ossxp-freemind (0.9.0~b20-14~wh) unstable; urgency=low

  * upstream upgrade to beta 20.

 -- Jiang Xin <worldhello.net@gmail.com>  Sat, 13 Sep 2008 22:14:46 +0800

ossxp-freemind (0.9.0~b19-14~wh) unstable; urgency=low

  * unescape utf8 entities in xml rich text node and note.

 -- Jiang Xin <worldhello.net@gmail.com>  Fri, 01 Aug 2008 22:18:39 +0800

ossxp-freemind (0.9.0~b19-13~wh) unstable; urgency=low

  * upstream upgrad;

 -- Jiang Xin <worldhello.net@gmail.com>  Fri, 01 Aug 2008 09:03:56 +0800

ossxp-freemind (0.9.0~b18-13~wh) unstable; urgency=low

  * upstream upgrad;

 -- Jiang Xin <worldhello.net@gmail.com>  Wed, 04 Jun 2008 20:30:09 +0800

ossxp-freemind (0.9.0~b17-13~wh) unstable; urgency=low

  * Change default properties:
    * set auto save time to 600 seconds;
    * set save_folding_state to false;
  * Add worldhello extension's setting dialog;

 -- Jiang Xin <worldhello.net@gmail.com>  Thu, 22 May 2008 23:25:50 +0800

ossxp-freemind (0.9.0~b17-12~wh) unstable; urgency=low

  * new resources strings translation.

 -- Jiang Xin <worldhello.net@gmail.com>  Thu, 22 May 2008 19:14:12 +0800

ossxp-freemind (0.9.0~b16-12~wh) unstable; urgency=low

  * new resources strings translation.

 -- Jiang Xin <worldhello.net@gmail.com>  Fri, 16 May 2008 21:16:25 +0800

ossxp-freemind (0.9.0~b16-11~wh) unstable; urgency=low

  * upstream upgrade

 -- Jiang Xin <worldhello.net@gmail.com>  Wed, 23 Apr 2008 19:44:35 +0800

ossxp-freemind (0.9.0~b9-11~wh) unstable; urgency=low

  * filter out all illegal xml characters when save xml file.

 -- Jiang Xin <worldhello.net@gmail.com>  Thu, 07 Jun 2007 08:40:29 +0800

ossxp-freemind (0.9.0~b9-10) unstable; urgency=low

  * bugfix for windows.
  * only save node in .mmx file.
  * use saxon as xslt process engine for freemind_join_mm_mmx.xslt.
  * use key() in freemind_join_mm_mmx.xslt. greate performance improved!

 -- Jiang Xin <worldhello.net@gmail.com>  Mon, 12 Mar 2007 12:00:18 +0800

ossxp-freemind (0.9.0~b9-5) unstable; urgency=low

  * store freemind flash in accessories/plugins/flash for other packages;
  * change algorithm of split and join .mm with .mmx;

 -- Jiang Xin <worldhello.net@gmail.com>  Sun, 11 Mar 2007 00:54:05 +0800

ossxp-freemind (0.9.0~b9-4) unstable; urgency=low

  * Orignal version saved as *.vanilla.*;
  * Use light-bulb splash screen;
  * add config files into /opt/freemind/conf;

 -- Jiang Xin <worldhello.net@gmail.com>  Fri, 09 Mar 2007 18:27:35 +0800

ossxp-freemind (0.9.0~b9-1) unstable; urgency=low

  * WorldHello.net hacked version;
  * Save output .mm file using Chinese utf-8;
  * Do not save nodes' folded status in to .mm file, save to .mmx file;

 -- Jiang Xin <worldhello.net@gmail.com>  Fri, 08 Mar 2007 10:16:31 +0800


freemind/freemind.sh
#!/bin/sh
# 2004-02-13, modified for Debian by deb@zorglub.s.bawue.de
# 2004-06-19, rewritten for Linux/UN*X by deb@zorglub.s.bawue.de
#             (based on Jan Schulz's input)
# 2004-11-28, minor changes for version 0.8.0 by deb@zorglub.s.bawue.de
# 2005-01-08, removed bashims to make script POSIX conform
# 2005-01-16, added usage of FREEMIND_BASE_DIR variable
# 2005-02-18, add -Dfreemind.base.dir to make plugins work, add some ""
#             and enhance debug mode.
# 2005-11-08, adding commons-codec to classpath.
# 2005-11-09, add some dpkg/rpm information and check for Sun/Blackdown VM.
# 2006-10-29, follow links to this script using readlink.
# 2008-02-02, improve Java recognition, add lsb_release, fix -x which being empty
#             add -Dgnu.java.awt.peer.gtk.Graphics=Graphics2D for non-Sun JREs
# 2008-02-03, add debug values script and exit
# 2008-05-20 (fc): -Xmx256M added

# we only want to test the script, not FreeMind itself
if ( echo "${DEBUG}" | grep -qe "script" )
then
	set -x
fi

########## FUNCTIONS DEFINITIONS #######################################

_debug() {
	if [ -n "${DEBUG}" ]
	then
		echo "DEBUG:   $1" >&2 
		shift
		for text in "$@"
		do
			echo "         ${text}" >&2
		done
	fi
}

_error() {
	echo "ERROR:   $1" >&2
	shift
	for text in "$@"
	do
		echo "         ${text}" >&2
	done
}

findjava() {
	# We try hard to find the proper 'java' command
	if [ -n "${JAVACMD}" ] && [ -x "${JAVACMD}" ]
	then
		_debug "Using \$JAVACMD to find java virtual machine."
	elif [ -n "${JAVA_BINDIR}" ] && [ -x "${JAVA_BINDIR}/java" ]
	then
		JAVACMD="${JAVA_BINDIR}/java"
		_debug "Using \$JAVA_BINDIR to find java virtual machine."
	elif [ -n "${JAVA_HOME}" ] && [ -x "${JAVA_HOME}/bin/java" ]
	then
		JAVACMD="${JAVA_HOME}/bin/java"
		_debug "Using \$JAVA_HOME to find java virtual machine."
	else
		JAVACMD=$(which java)
		if [ -n "${JAVACMD}" ] && [ -x "${JAVACMD}" ]
		then
			_debug "Using \$PATH to find java virtual machine."
		elif [ -x /usr/bin/java ]
		then
			_debug "Using /usr/bin/java to find java virtual machine."
			JAVACMD=/usr/bin/java
		fi
	fi

	# if we were successful, we return 0 else we complain and return 1
	if [ -n "${JAVACMD}" ] && [ -x "${JAVACMD}" ]
	then
		_debug "Using '$JAVACMD' as java virtual machine..."
		if [ -n "${DEBUG}" ]
		then
			"$JAVACMD" -version >&2
		fi
		if (! "${JAVACMD}" -version 2>&1 | grep -qe 'Java(TM)')
		then
			_error "Your Java is not a derivative from Sun's code," \
			       "=======================================" \
			       "FREEMIND WILL MOST PROBABLY *NOT* WORK," \
			       "=======================================" \
			       "define JAVACMD, JAVA_BINDIR, JAVA_HOME or PATH in order" \
			       "to point to such a VM. See the manpage of freemind(1) for details."
			JAVA_TYPE=other
		else
			JAVA_TYPE=sun
		fi
		return 0
	else
		_error "Couldn't find a java virtual machine," \
		       "define JAVACMD, JAVA_BINDIR, JAVA_HOME or PATH."
		return 1
	fi
}

_source() {
	if [ -f "$1" ]
	then
		_debug "Sourcing '$1'."
		. "$1"
	fi
}

output_debug_info() {
	if [ -z "${DEBUG}" ]
	then
		return 0
	fi
	_debug "Freemind parameters are '${@}'."
	_debug "$(uname -a)"
	if [ -x "$(which lsb_release 2>/dev/null)" ]
	then
		_debug "$(lsb_release -a)"
	else
		_debug "System is not LSB conform, 'lsb_release' does not exist."
	fi
	if [ -x "$(which dpkg 2>/dev/null)" ]
	then
		_debug "The following DEB packages are installed:"
		COLUMNS=132 dpkg -l | grep -i -e freemind >&2
	elif [ -x "$(which rpm 2>/dev/null)" ]
	then
		_debug "The following RPM packages are installed:"
		rpm -qa | grep -i -e freemind >&2
	else
		_debug "Neither dpkg nor rpm is installed."
	fi
}

########## START MAIN PART #############################################

#--------- Put the environment together --------------------------------

_source /etc/freemind/freemindrc
_source ~/.freemind/freemindrc

findjava
if [ $? -ne 0 ]
then
	exit 1
fi

output_debug_info

if [ -L "$0" ] && [ -x $(which readlink) ]
then # if the script is a link and we have 'readlink' to follow it
	# -m should be faster and link does always resolve, else this script
	# wouldn't be called, would it?
	freefile=$(readlink -mn "$0")
	_debug "Link '$0' resolved to '${freefile}'."
else
	freefile="$0"
fi
freepath=$(dirname "${freefile}")
freepath="${freepath%/bin}" # nothing happens if freemind is not installed
                            # under something/bin

# we try different possibilities to find freemind.jar
for jar in "${FREEMIND_BASE_DIR}" \
	"${freepath}" "${freepath}/share/freemind" "${freepath}/freemind"
do
	if [ -f "${jar}/lib/freemind.jar" ]
	then
		freedir="${jar}"
		_debug "Freemind Directory is '${jar}'."
		break
	fi
done

if [ -z "${freedir}" ]
then
	_error "Couldn't find freemind under '${freepath}'."
	exit 1
fi

if [ ! -f ~/.freemind/patterns.xml ] && [ -f /etc/freemind/patterns.xml ]
then
	if [ ! -d ~/.freemind ]
	then
		_debug "Creating directory ~/.freemind."
		mkdir -p ~/.freemind
	fi
	_debug "Copying patterns.xml to ~/.freemind."
	cp /etc/freemind/patterns.xml ~/.freemind/patterns.xml
fi

#--------- Call (at last) FreeMind -------------------------------------

# The CLASSPATH also lets one specify additional jars, which is good, if
# you want to add a new Look&Feel jar (the motif one is so ugly...).
# 
CLASSPATH="${ADD_JARS}:${CLASSPATH}:${freedir}/lib/freemind.jar:\
${freedir}/lib/jibx/jibx-run.jar:\
${freedir}/lib/jibx/xpp3.jar:\
${freedir}/lib/bindings.jar:\
${freedir}/lib/commons-lang-2.0.jar:\
${freedir}/lib/forms-1.0.5.jar:\
${freedir}"
if [ "${JAVA_TYPE}" = "sun" ]
then
	_debug "Calling: '${JAVACMD} -Dfreemind.base.dir=${freedir} -cp ${CLASSPATH} freemind.main.FreeMindStarter  $@'."
	( echo "${DEBUG}" | grep -qe "exit" ) && exit 0 # do not start FreeMind
	"${JAVACMD}" -Xmx256M -Dfreemind.base.dir="${freedir}" -cp "${CLASSPATH}" freemind.main.FreeMindStarter "$@"
else # non-Sun environments don't work currently.
	_debug "Calling: '${JAVACMD} -Dgnu.java.awt.peer.gtk.Graphics=Graphics2D -Dfreemind.base.dir=${freedir} -cp ${CLASSPATH} freemind.main.FreeMindStarter  $@'."
	( echo "${DEBUG}" | grep -qe "exit" ) && exit 0 # do not start FreeMind
	"${JAVACMD}" -Xmx256M -Dgnu.java.awt.peer.gtk.Graphics=Graphics2D -Dfreemind.base.dir="${freedir}" -cp "${CLASSPATH}" freemind.main.FreeMindStarter "$@"
fi
